### PR TITLE
No warning for signal_psd: default min_freq

### PR DESCRIPTION
**Pull request by Tam-Pham**
warning for low-frequency resolution
Address #560 

# Checklist

Here are some things to check before creating the PR. If you encounter any issues, do let us know :)

- [x] I have read the [CONTRIBUTING](https://github.com/neuropsychology/NeuroKit/blob/master/.github/CONTRIBUTING.rst#structure-and-code) file.
- [x] My PR is targetted at the **dev branch** (and not towards the master branch).
- [ ] I ran the [CODE CHECKS](https://github.com/neuropsychology/NeuroKit/blob/master/.github/CONTRIBUTING.rst#run-code-checks) on the files I added or modified and fixed the errors.
- [ ] I have added the newly added features to **News.rst** (if applicable)

### CODE REVIEW

**Comment by DominiqueMakowski** [comment]
But so you agree that currently, by default, in most of the cases: `nperseg = int(len(signal) / 2)` right?

**Comment by DominiqueMakowski** [comment]
why not "sanitize" min_freq to the lowest possible instead of 0.001

**Comment by DominiqueMakowski** [comment]
doing: `min_freq = max(min_freq, min_possible_freq_given_length_and_SR)` and then padding from 0 ?

**Comment by DominiqueMakowski** [comment]
We can sanitize min_freq like so, assuming it is 0 by default:

`min_freq = max(min_freq, 1/(len(signal) / sampling_rate))`

**Comment by Tam-Pham** [comment]
> But so you agree that currently, by default, in most of the cases: nperseg = int(len(signal) / 2) right?

No 😅 only when the recording length is not sufficiently long to capture at least 2 cycles of min_frequency

I don't know if you are misunderstanding but there is no "impossible frequencies". We are just using the min_frequency to determine the length of overlapping segments when calculating PSD.

**Comment by DominiqueMakowski** [comment]
>  only when the recording length is not sufficiently long

How come i get the warning *everytime* I use signal_psd then??

**Comment by Tam-Pham** [comment]
> How come i get the warning _everytime_ I use signal_psd then?? even for signals of length a million

Because if min_frequency is zero (which is adjusted to 0.01), the signal length required for "absolute" high level of good resolution is 4 million 😅 

For your use case, you don't have a default value of min_frequency that you want?

**Comment by DominiqueMakowski** [comment]
hence the current "adjusted" value is useless since it requires a very rare usecase

All I want is a default behaviour that 1) doesnt' require to manually input min_freq and 2) doesn't print a warning for 99.999% of cases 😅

**Comment by Tam-Pham** [comment]
> All I want is a default behaviour that 1) doesnt' require to manually input min_freq and 2) doesn't print a warning for 99.999% of cases 😅

What we can do is add a "default" option for min_frequency where min_frequency is determined by optimal nperseg (based on sampling_rate and length of the signal.

**Comment by DominiqueMakowski** [comment]
> What we can do is add a "default" option

File: neurokit2/complexity/entropy_multiscale.py [hunk 1]
```diff
@@ -250,6 +244,15 @@ def _entropy_multiscale_cmse(signal, tau, dimension, tolerance, fuzzy, **kwargs)
             y[i, :], delay=1, dimension=dimension, tolerance=tolerance, fuzzy=fuzzy, **kwargs
         )[0]
 
+    if len(np.where((mse_y == np.inf) | (mse_y == -np.inf) | (mse_y == np.nan))[0]) == len(mse_y):
+        # return nan if all are infinity/nan values
+        return np.nan
+    else:
+        # Remove inf, nan and 0
+        mse_y = mse_y[~np.isnan(mse_y)]
```
Comment (DominiqueMakowski) [inline neurokit2/complexity/entropy_multiscale.py:252]: can't do one liner? `mse_y[(~np.isnan(mse_y)) | (mse_y != np.inf) | (mse_y != -np.inf)]`

Comment (zen-juen) [reply neurokit2/complexity/entropy_multiscale.py]: u right ^^

**Comment by DominiqueMakowski** [comment]
@Tam-Pham currently we "normalize" the power in many different places (in signal_psd), can we unify that so that it's done only at one place?

**Comment by DominiqueMakowski** [comment]
Why this so nice:

```python
import neurokit2 as nk
import mne
raw.plot_psd(fmin=0, fmax=40., picks=["EEG 050"])
```

image


And ours so ugly 🥲 

```python
channel = nk.mne_channel_extract(raw, what=["EEG 050"]).values
psd = nk.signal_psd(
    channel, sampling_rate=raw.info["sfreq"], show=True, max_frequency=40, method="multitapers"
)
```

image

**Comment by DominiqueMakowski** [comment]
> And ours so ugly 🥲

what to do lah

**Comment by Tam-Pham** [comment]
Next PR? 🥲

**Pull request merged**